internal/controlplane/handlers_oauth.go

```diff
@@ -290,6 +290,18 @@ func (s *Server) processOAuthCallback(ctx context.Context, w http.ResponseWriter
 	if err != nil {
 		return fmt.Errorf("error encoding token: %w", err)
 	}
+	// Verify if the token user is a member of the ownerFilter (Organization)
+	if stateData.OwnerFilter.Valid {
+		member, err := s.ghProviders.ValidateOrgMembershipForToken(ctx, token, stateData.OwnerFilter.String)
+		if err != nil {
+			return fmt.Errorf("error checking organization membership: %w", err)
+		}
+
+		if !member {
+			return newHttpError(http.StatusForbidden, "Invalid OwnerFilter Provided").SetContents(
+				"The ownerFilter value provided does not allow access for the logged in user")
```

Comment:
After the refactor, I noticed we will be returning the same error message incase of invalid org or any github API failures. Do you think we should handle the github API failure separately ?

Reply:
Instead of returning the boolean from the helper function, we can just return the error. Then the caller can directly return the error as well. This way, we can keep the error messages for invalid org or GitHub API failures.
It's fairly common to do that in our codebase, for example:
https://github.com/stacklok/minder/blob/16421648e284c9f5be7c5261aaa8e5a94b61c70c/internal/controlplane/handlers_oauth.go#L345-L348